azavea/climate-change-lab
# Feature/awf/angular cli

## Overview

Updates the projects Angular dependencies, to 4.0+, and replaces the custom webpack build with the Angular CLI, now that it is out of beta.

### Demo

Not really, app looks the same.


### Notes

The Angular CLI upgrade was relatively straightforward, but there was a lot of stuff to fix. I generated a new project outside the repo via `ng new` then copied the relevant CLI files into the project and deleted the old ones. Then cleaned up a bunch of changed file paths and broken stuff from the change.

## Testing Instructions

- `rm -rf node_modules`
- `npm install`
- `npm start`

The dev server should start and reload on file change. 

Test prod build:
- `npm run build:prod`
- `http-server -p 7777 dist/`

Load http://localhost:7777 should see the app working, with a much smaller total asset download (~1.2mb)

`npm run test` should complete successfully.

`npm run e2e` will fail because no e2e tests are defined.

`npm run lint` currently fails. ~~Going to spend some time attempting to clean this up as a separate commit on this PR later.~~ <-- EDIT: ~~The changes are massive and would detract from the changes made to update to Angular CLI. Will open as a separate PR.~~ See #143 


Closes #139, Closes #140
